### ISSUE TITLE
FFMPEG Issues

### ISSUE THREAD

**Issue by zombielinux** [mmcc-xx/BirdCAGE#33]
I'm getting the error: 

"rtmp://fqdn/bcs/channel0_main.bcs?user=admin&password=$PASS: No such file or directory"
"Error: ffmpeg error (see stderr output for detail)"
" Recording failed. Error: ffmpeg error (see stderr output for detail)"

How can I redirect the stderr to the docker log?

Further, I had similar issues with this (Reolink) camera in Frigate and had to add specific ffmpeg directives. Specifically:
```
    ffmpeg:
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer
        - -flags
        - low_delay
        - -strict
        - experimental
        - -fflags
        - +genpts+discardcorrupt
        - -rw_timeout
        - '5000000'
        - -use_wallclock_as_timestamps
        - '1'
```

From my frigate config.

**Comment by WanderingStar** [comment]
I submitted a PR to print the contents of stderr. It doesn't really help in my case, because it just says

```
  stderr: alsa_input.pci-0000_00_1f.3.analog-stereo.2: Input/output error
```

**Comment by mmcc-xx** [comment]
Ahh see the problem is that you've got an Input/output error. ffmpeg can be so helpful.

Thanks for the PR - I merged it in. 

There might be a way to increase the loglevel to get ffmpeg to give more info:

https://github.com/kkroening/ffmpeg-python/issues/30

I have not done more than briefly skim that, but it does mention a way to pass in loglevel.

**Comment by WanderingStar** [comment]
I ended up using `docker exec` and running `pactl` and `ffmpeg` in the container until I got it to work. I think commenting this out of `default.pa` is the thing that made it finally work:
```load-module module-suspend-on-idle``` 

But now the logs show it's recording audio and analyzing it to have vanishingly small probabilities of birds even when I play bird sounds right into the mic, so further debugging to do.

Incidentally, is there a way to let people PR edits to the wiki?

**Comment by mmcc-xx** [comment]
> Incidentally, is there a way to let people PR edits to the wiki?

I have no idea :) With it being a wiki I'd assume you can just change stuff willy-nilly. What did you have in mind?

When I did my testing of pulse audio, I played bird songs from youtube into a mic and it seemed to work. But, audio in linux always makes me crazy.

**Comment by WanderingStar** [comment]
I don't know how much of spam magnet this is, but: https://docs.github.com/en/communities/documenting-your-project-with-wikis/changing-access-permissions-for-wikis

I was going to add some tips on what helped me to debug the PA problems.

**Comment by mmcc-xx** [comment]
Yeah I don't think I want to open it up - would you object to being made a collaborator?

**Comment by WanderingStar** [comment]
No, I'm happy to be a collaborator.
